test/FakeStateMachine-runState.ts

```diff
@@ -184,14 +303,228 @@ describe('FakeStateMachine#runState()', () => {
+            await fakeStateMachine.runState(
+              { condition: 0 },
+              'Choices'
+            )
+          ).toEqual(
+            new RunStateResult(
+              { condition: 0 },
+              'Choice',
+              'DefaultState',
+              false
+            )
+          );
+        });
       });
       describe('with Or', () => {
-        test.todo('pending');
+        const choice: TopLevelChoiceRule = {
+          Or: [
+            {
+              Variable: '$.condition',
+              NumericLessThan: 4,
+            },
+            {
+              Variable: '$.condition',
+              NumericGreaterThan: 5,
+            },
+          ],
+          Next: 'NextState',
+        };
+
+        const definition = {
+          States: {
+            Choices: {
+              Type: 'Choice',
+              Choices: [choice],
+              Default: 'DefaultState',
+            },
+          },
+        };
+
+        test(`the next state should be NextState`, async () => {
+          const fakeStateMachine = new FakeStateMachine(definition, {});
+          expect(
+            await fakeStateMachine.runState(
+              { condition: 4.5 },
+              'Choices'
+            )
+          ).toEqual(
+            new RunStateResult(
+              { condition: 4.5 },
+              'Choice',
+              'DefaultState',
+              false
+            )
+          );
+        });
+
+        test(`the next state should be NextState`, async () => {
+          const fakeStateMachine = new FakeStateMachine(definition, {});
+          expect(
+            await fakeStateMachine.runState(
+              { condition: 5.5 },
+              'Choices'
+            )
+          ).toEqual(
+            new RunStateResult(
+              { condition: 5.5 },
+              'Choice',
+              'NextState',
+              false
+            )
+          );
+        });
+
+        test(`the next state should be NextState`, async () => {
+          const fakeStateMachine = new FakeStateMachine(definition, {});
+          expect(
+            await fakeStateMachine.runState(
+              { condition: 0 },
+              'Choices'
+            )
+          ).toEqual(
+            new RunStateResult(
+              { condition: 0 },
+              'Choice',
+              'NextState',
+              false
+            )
+          );
+        });
       });
       describe('with Not', () => {
-        test.todo('pending');
+        const choice: TopLevelChoiceRule = {
+          Not: {
+            Variable: '$.condition',
+            NumericLessThan: 5,
+          },
+          Next: 'NextState',
+        };
+
+        const definition = {
+          States: {
+            Choices: {
+              Type: 'Choice',
+              Choices: [choice],
+              Default: 'DefaultState',
+            },
+          },
+        };
+
+        test(`the next state should be NextState`, async () => {
```

Comment:
This case should be `DefaultState`?
```suggestion
        test(`the next state should be DefaultState`, async () => {
```

Reply:
Fixed.